pkg/config/kubeclient.go

```diff
@@ -0,0 +1,99 @@
+package config
+
+import (
+	"sync"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/informers"
+	v1informers "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/openshift/baremetal-runtimecfg/pkg/utils"
+)
+
+type KubeClient struct {
+	*kubernetes.Clientset
+	sync.Mutex
+
+	factory      informers.SharedInformerFactory
+	nodeInformer v1informers.NodeInformer
+	stopCh       <-chan struct{}
+}
+
+// NewKubeClient returns a KubeClient for the given config
+//
+// Args:
+//   - api-server URL as string (optional)
+//   - kubeconfigPath as string
+//   - stopCh as a stop channel
+//
+// Returns:
+//   - KubeClient or error
+func NewKubeClient(apiserverURL, kubeconfigPath string, stopCh <-chan struct{}) (*KubeClient, error) {
+	config, err := utils.GetClientConfig(apiserverURL, kubeconfigPath)
+	if err != nil {
+		log.WithFields(logrus.Fields{
+			"err":    err,
+			"path":   kubeconfigPath,
+			"apiurl": apiserverURL,
+		}).Error("Failed to get client config")
+		return nil, err
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.WithFields(logrus.Fields{
+			"err": err,
+		}).Error("Failed to get client")
+		return nil, err
+	}
+
+	return &KubeClient{
+		Clientset: kubeClient,
+		stopCh:    stopCh,
+	}, err
+}
+
+// Start starts a KubeClient
+func (c *KubeClient) Start() {
+	c.Lock()
+	if c.factory == nil {
+		c.factory = informers.NewSharedInformerFactory(c, 0)
+		c.nodeInformer = c.factory.Core().V1().Nodes()
+	}
+	c.Unlock()
+
+	c.factory.Start(c.stopCh)
+	if ok := cache.WaitForCacheSync(c.stopCh, c.nodeInformer.Informer().HasSynced); !ok {
```

Comment:
The master nodes seem to be stuck on this line in my local env. I'm not clear why since the boostrap API should be up and holding the VIP at the time, so we should be able to talk to it. :-/